db_refactor/instance_schema.sql

```diff
@@ -0,0 +1,733 @@
+  state policy_state NOT NULL,
+  user_login_must_be_domain BOOLEAN NOT NULL,
+  validate_org_domains BOOLEAN NOT NULL,
+  smtp_sender_address_matches_instance_domain BOOLEAN NOT NULL,
+  is_default BOOLEAN DEFAULT FALSE NOT NULL,
+  created_at TIMESTAMP DEFAULT NOW(),
+  updated_at TIMESTAMP DEFAULT NOW(),
+  deleted_at TIMESTAMP DEFAULT NULL
+);
+
+-- Password/Security
+-----------------------------------------------------------------------------------------------------------------------------------------------------------
+-- TODO consider combining lockout_policies + password_complexity_policies + password_age_policies
+-- Lockout Policies
+DROP TABLE IF EXISTS zitadel.lockout_policies CASCADE;
+CREATE TABLE zitadel.lockout_policies (
+  id SERIAL NOT NULL PRIMARY KEY,
+  instance_id VARCHAR(100) NOT NULL,
+  CONSTRAINT instance_id_fk FOREIGN KEY(instance_id) REFERENCES instances(id) ON DELETE CASCADE,
+  state policy_state NOT NULL,
+  is_default BOOLEAN DEFAULT FALSE NOT NULL,
+  max_password_attempts BIGINT NOT NULL,
+  max_otp_attempts BIGINT DEFAULT 0 NOT NULL,
+  show_failure BOOLEAN NOT NULL,
+  created_at TIMESTAMP DEFAULT NOW(),
+  updated_at TIMESTAMP DEFAULT NOW(),
+  deleted_at TIMESTAMP DEFAULT NULL
+);
+
+-- Password Age
+DROP TABLE IF EXISTS zitadel.password_age_policies CASCADE;
+CREATE TABLE zitadel.password_age_policies (
+  id SERIAL NOT NULL PRIMARY KEY,
+  instance_id VARCHAR(100) NOT NULL,
+  CONSTRAINT instance_id_fk FOREIGN KEY(instance_id) REFERENCES instances(id) ON DELETE CASCADE,
+  state policy_state NOT NULL,
+  is_default BOOLEAN DEFAULT FALSE NOT NULL,
+  expire_warn_days BIGINT NOT NULL,
+  max_age_days BIGINT NOT NULL,
+  created_at TIMESTAMP DEFAULT NOW(),
+  updated_at TIMESTAMP DEFAULT NOW(),
+  deleted_at TIMESTAMP DEFAULT NULL
+);
+
+-- Password Complexity
+DROP TABLE IF EXISTS zitadel.password_complexity_policies CASCADE;
+CREATE TABLE zitadel.password_complexity_policies (
+  id SERIAL NOT NULL PRIMARY KEY,
+  instance_id VARCHAR(100) NOT NULL,
+  CONSTRAINT instance_id_fk FOREIGN KEY(instance_id) REFERENCES instances(id) ON DELETE CASCADE,
+  state policy_state NOT NULL,
+  is_default BOOLEAN DEFAULT FALSE NOT NULL,
+  min_length BIGINT NOT NULL,
+  has_lowercase BOOLEAN NOT NULL,
+  has_uppercase BOOLEAN NOT NULL,
+  has_symbol BOOLEAN NOT NULL,
+  has_number BOOLEAN NOT NULL,
+  created_at TIMESTAMP DEFAULT NOW(),
+  updated_at TIMESTAMP DEFAULT NOW(),
+  deleted_at TIMESTAMP DEFAULT NULL
+);
+
+-- idp
+-----------------------------------------------------------------------------------------------------------------------------------------------------------
+DROP TYPE IF EXISTS zitadel.idp_type CASCADE;
+CREATE TYPE zitadel.idp_type AS ENUM ('oauth2', 'oidc', 'jwt', 'azure', 'github', 'gitlab', 'github_enterprise', 'gitlab_self_hosted', 'google', 'ldap', 'saml', 'apple');
+DROP TABLE IF EXISTS zitadel.idp_providers CASCADE;
+CREATE TABLE zitadel.idp_providers (
```

Comment:
idp_provider is redundant (identity_provider_providers) i would go with either idps or identity_providers